sandbox/modules/identity/src/main/java/org/opensearch/identity/IdentityPlugin.java

```diff
@@ -268,6 +324,9 @@ public Collection<Object> createComponents(
 
         cr.setDynamicConfigFactory(dcf);
 
+        IdentityJwtVerifier verifier = IdentityJwtVerifier.getInstance();
+        verifier.init(this.settings.get(IdentityConfigConstants.IDENTITY_SIGNING_KEY));
```

Comment:
For my understanding, why is a verification needed here?

Reply:
This is initializing this singleton of the `IdentityJwtVerifier` with the configured signing key. I needed to do it this way, because I was struggling to figure out a way to get the same instance of this verifier into the `InternalRealm` to verify bearer tokens. The `IdentityJwtVerifier` is the verifier used for bearer auth for authenticating to the cluster from a client. Opposed, to a general `JwtVerifier` which is a class used to make new instances of a verifier from a signing key. This is used to create extension specific verifiers from the extensions secret signing key.

Reply:
ahh..got it. Seems like IdentityJwtVerifier usage is limited. Can you also add a bried javadoc for both classes stating the same?

Reply:
Added